tests/rustdoc-ui/doctest/nocapture-fail.stderr

```diff
@@ -14,5 +14,12 @@ LL |     Input: 123
 LL ~ } }
    |
 
-error: aborting due to 1 previous error
+error[E0601]: `main` function not found in crate `rust_out`
```

Comment:
I consider this an unfortunate regression. As mentioned in my other review comment, this doctest no longer gets preprocessed (like adding a `main` if absent).

If you like to see this addressed, I can of course change it to not bail out on `has_errors` (but still set `failed_to_parse` to `true`) in order to forward `has_main` and so correctly.

However in general (not necessarily in this specific case), wrapping invalid doctests in a function is always a trade-off (it may lead to better errors in some cases but worse in others; and it may leak internal names in diagnostics like `_doctest_main_tests_rustdoc_ui_doctest_comment_in_attr_134221_2_rs_11_0`; see https://github.com/rust-lang/rust/issues/142446#issuecomment-3445358460 for context).